### PR TITLE
XOP-681: Fix some concurrency issue on host patching

### PR DESCRIPTION
**Pull request by zli**
Tests done:

* run all the patching related sequences in nightly to verify there is no regression
* manually created several scenarios where concurrency is an issue and verified the change can fix these

### CODE REVIEW

**Comment by johnelse** [comment]
@xen-git test this please

File: ocaml/xapi/xapi_pool_patch.ml [hunk 1]
```diff
@@ -403,13 +403,23 @@ let get_patch_to_local ~__context ~self =
...
 
-let patch_is_applied_to ~__context ~patch ~host =
-  let expr = 
-    And (Eq (Field "pool_patch", Literal (Ref.string_of patch)), 
-         Eq (Field "host",  Literal (Ref.string_of host))) 
-  in
-  let result = Db.Host_patch.get_records_where ~__context ~expr in
-    (List.length result) > 0
+let patch_apply_in_progress ~__context ~patch ~host =
+	let message = Printf.sprintf "Applying the same patch on %s is in progress. If you believe this is not true or the patching process is hang/freeze, try to restart the toolstack or host." (Db.Host.get_name_label ~__context ~self:host) in
```
Comment (johnelse) [inline ocaml/xapi/xapi_pool_patch.ml]: I would word this message as "the patching process is hung/frozen".

**Comment by johnelse** [comment]
Code looks good to me!

File: ocaml/xapi/xapi_pool_patch.ml (hunk 1 above)
Comment (zli) [inline ocaml/xapi/xapi_pool_patch.ml]: Yep, thanks for the correction. Updated.

**Pull request merged**